BB-624 Retry connection to kafka if backbeat processes started before

BourgoisMickael opened the pull request:
- Suppress AWS SDK warning in stderr
- Improve container image size from 2.34GB to 1.55GB (layer chown + dockerignore)
```bash
REPOSITORY                   TAG                         IMAGE ID       SIZE         BLOB SIZE
scality/backbeat             7.70.15.rc1-nodesvc-base    ea2e98003485    1.6 GiB      491.7 MiB
scality/backbeat             9.4.0.0                     cf4ecf197272    2.3 GiB      1.8 GiB

```
- Backport / cherrypick some fixes from zenko to ~~crash~~ exit after a timeout on connection error with kafka client (for CRR, Lifecycle, Bucket notification). Even for bucket notification destination.
- Prevent ProbeServer crash on startup by fixing metric handling when the consumer is not set up and delaying it after the queue components are started.


New behavior: components will try to connect for 60s instead of 30s and ~~crash~~ exit and restart if kafka client can't connect with error like:
```javascript
{"name":"Backbeat:QueueProcessor:task","time":1730775546353,"method":"QueueProcessor::task","error":{"origin":"local","message":"broker transport failure","code":-195,"errno":-195,"stack":"Error: Local: Broker transport failure\n    at Function.createLibrdkafkaError [as create] (/home/scality/backbeat/node_modules/node-rdkafka/lib/error.js:454:10)\n    at /home/scality/backbeat/node_modules/node-rdkafka/lib/client.js:350:28"},"level":"error","message":"error during queue processor initialization","hostname":"MDM-RING-46789-store-1","pid":594}
```
```javascript
{"name":"Backbeat:QueueProcessor:task","time":1730776197182,"error":{"origin":"local","message":"timed out","code":-185,"errno":-185,"stack":"Error: Local: Timed out\n    at Function.createLibrdkafkaError [as create] (/home/scality/backbeat/node_modules/node-rdkafka/lib/error.js:454:10)\n    at /home/scality/backbeat/node_modules/node-rdkafka/lib/client.js:350:28"},"method":"MetricsProducer::setupProducer","level":"error","message":"error starting metrics producer for queue processor","hostname":"MDM-RING-46789-store-2","pid":146}
```

giacope committed:
bf: crash when unable to connect with kafka

(cherry picked from commit 101af70c242cd7aec100c2f3592a914f4b672f2c)

giacope committed:
bf: crash backbeat consumer when unable to connect with kafka

(cherry picked from commit 22282ac2f0a38515ae0e32ebb1923206e56eb778)

giacope committed:
bf: crash backbeat components when unable to connect to kafka

(cherry picked from commit 3c2aa762027861caae6c061ce0c78dddf32a64a5)

BourgoisMickael committed:
BB-624: Suppress AWS SDK v3 warning in stderr

Missing from S3C-9338 because this usage of
sdk doesn't use the env variable

BourgoisMickael committed:
BB-624: Improve image img size from 2.34G to 1.55G

Will be improved in S3C-9479

BourgoisMickael committed:
BB-624: S3C specific handling of startup failure

bert-e commented:
# Hello bourgoismickael,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-624 contains:


* `7.10.18`

* `7.70.15`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.18`

* `7.70.15`

* `8.5.5`

* `8.6.50`

* `8.7.0`


Please check the `Fix Version/s` of BB-624, or the target
branch of this pull request.

BourgoisMickael committed:
BB-624: backport some notification startup fix

Copied from eb1e608766fecaf52ac2710345656a8bb2ed8b67

BourgoisMickael commented:
/create_integration_branches

BourgoisMickael commented:
ping

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.5/bugfix/BB-624-connection-retry` with contents from `w/7.70/bugfix/BB-624-connection-retry`
and `development/8.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.5/bugfix/BB-624-connection-retry origin/development/8.5
 $ git merge origin/w/7.70/bugfix/BB-624-connection-retry
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.5/bugfix/BB-624-connection-retry
```





*The following options are set:* **create_integration_branches**

BourgoisMickael committed:
BB-624 S3C-9436: Prevent startup crash on probing

If probing too soon on startup some CRR components
can crash.
Check consumer exists for metric function to use it.
Also the probe server is started after the queue (like in ZENKO)

BourgoisMickael committed:
BB-624: Don't ignore setup error for CRR status

Prevent hanging indefinitely if replication status' BackbeatConsumer
succeeds to connect to kafka but then FailedCRRPRoducer or
ReplayProducer fails

BourgoisMickael commented:
ping

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-624-connection-retry` into
`development/7.10`
* [w/7.70/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/7.70...w/7.70/bugfix/BB-624-connection-retry) will be merged into `development/7.70` 
* [w/8.5/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.5...w/8.5/bugfix/BB-624-connection-retry) will be merged into `development/8.5` 
* [w/8.6/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.6...w/8.6/bugfix/BB-624-connection-retry) will be merged into `development/8.6` 
* [w/8.7/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.7...w/8.7/bugfix/BB-624-connection-retry) will be merged into `development/8.7` 



The following branches will **NOT** be impacted:

* `development/7.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

BourgoisMickael commented:
@bert-e create_pull_requests

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-624-connection-retry` into
`development/7.10`
* [w/7.70/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/7.70...w/7.70/bugfix/BB-624-connection-retry) will be merged into `development/7.70` (pull request #2577)
* [w/8.5/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.5...w/8.5/bugfix/BB-624-connection-retry) will be merged into `development/8.5` (pull request #2578)
* [w/8.6/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.6...w/8.6/bugfix/BB-624-connection-retry) will be merged into `development/8.6` (pull request #2579)
* [w/8.7/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.7...w/8.7/bugfix/BB-624-connection-retry) will be merged into `development/8.7` (pull request #2580)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/7.70/bugfix/BB-624-connection-retry` with contents from `bugfix/BB-624-connection-retry`
and `development/7.70`.

Please resolve the conflict on **the integration branch** (`w/7.70/bugfix/BB-624-connection-retry`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/7.70/bugfix/BB-624-connection-retry
 $ git pull  # or "git reset --hard origin/w/7.70/bugfix/BB-624-connection-retry"
 $ git merge origin/development/7.70
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/bugfix/BB-624-connection-retry
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.70/bugfix/BB-624-connection-retry
```





*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/8.5/bugfix/BB-624-connection-retry` with contents from `w/7.70/bugfix/BB-624-connection-retry`
and `development/8.5`.

Please resolve the conflict on **the integration branch** (`w/8.5/bugfix/BB-624-connection-retry`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/8.5/bugfix/BB-624-connection-retry
 $ git pull  # or "git reset --hard origin/w/8.5/bugfix/BB-624-connection-retry"
 $ git merge origin/development/8.5
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/7.70/bugfix/BB-624-connection-retry
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.5/bugfix/BB-624-connection-retry
```





*The following options are set:* **create_pull_requests, create_integration_branches**

nicolas2bert commented on bin/queuePopulator.js:
```diff
@@ -52,6 +52,7 @@ const queuePopulator = new QueuePopulator(
     zkConfig, kafkaConfig, qpConfig, httpsConfig, mConfig, rConfig, extConfigs);
```
Nit: commit title should be probing

nicolas2bert commented on extensions/notification/queueProcessor/QueueProcessor.js:
```diff
@@ -142,7 +142,15 @@ class QueueProcessor extends EventEmitter {
+                    { method: 'QueueProcessor.start', error: err.message });
+                    // crash if got error at startup
+                    if (!this.isReady()) {
+                        return next(err);
```
how would it crash here?

BourgoisMickael replied:
I've got callback error because I've not copied correctly from zenko branch. I'll fix that.

~~Then it's not a crash, it shoud print an error log and after there is no more instructions, if no handles / connection retain the process active, it should stop with exit code 0.~~

I'll add a process.exit(1) then like it's done in zenko

nicolas2bert commented on lib/BackbeatConsumer.js:
```diff
@@ -186,7 +186,18 @@ class BackbeatConsumer extends EventEmitter {
+            if (err) {
+                this.emit('error', err);
+            }
+            // const opts = { topic: 'backbeat-sanitycheck', timeout: 10000 };
```
do we really need that?

BourgoisMickael replied:
I'll remove

nicolas2bert commented on lib/BackbeatConsumer.js:
```diff
@@ -186,7 +186,18 @@ class BackbeatConsumer extends EventEmitter {
         }
         this._consumer = new kafka.KafkaConsumer(consumerParams);
-        this._consumer.connect();
+        // For S3C 60s + prevent any side effect with sanitycheck topic
```
adjust the comment.

nicolas2bert reviewed (commented):
Suggestions:
- add tests for "Prevent ProbeServer crash on startup by delaying it after the queue components are started".
- add tests for Consumer service crashes in the event of a consumer error?

BourgoisMickael committed:
BB-624: Notification restart if destination down

If destination is down on start, QueueProcessor would ignore
and start connecting to consumer.

Now if destination is down the process will stop and restart
so it can try connecting again to destination.

bert-e commented:
# History mismatch



Merge commit #8976f2649bb3df34e816a7d2cec2220e31246e47 on the integration branch
`w/7.70/bugfix/BB-624-connection-retry` is merging a branch which is neither the current
branch `bugfix/BB-624-connection-retry` nor the development branch
`development/7.70`.

It is likely due to a rebase of the branch `bugfix/BB-624-connection-retry` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests, create_integration_branches**

BourgoisMickael committed:
BB-624: Test CRR metrics at startup crash

BourgoisMickael commented:
reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**





*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #8976f2649bb3df34e816a7d2cec2220e31246e47 on the integration branch
`w/7.70/bugfix/BB-624-connection-retry` is merging a branch which is neither the current
branch `bugfix/BB-624-connection-retry` nor the development branch
`development/7.70`.

It is likely due to a rebase of the branch `bugfix/BB-624-connection-retry` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #fe920991b636b14b4ac1bcf2855aad317244e48b on the integration branch
`w/8.5/bugfix/BB-624-connection-retry` is merging a branch which is neither the current
branch `bugfix/BB-624-connection-retry` nor the development branch
`development/8.5`.

It is likely due to a rebase of the branch `bugfix/BB-624-connection-retry` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests, create_integration_branches**

BourgoisMickael commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.70/bugfix/BB-624-connection-retry` with contents from `bugfix/BB-624-connection-retry`
and `development/7.70`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.70/bugfix/BB-624-connection-retry origin/development/7.70
 $ git merge origin/bugfix/BB-624-connection-retry
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.70/bugfix/BB-624-connection-retry
```





*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.5/bugfix/BB-624-connection-retry` with contents from `w/7.70/bugfix/BB-624-connection-retry`
and `development/8.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.5/bugfix/BB-624-connection-retry origin/development/8.5
 $ git merge origin/w/7.70/bugfix/BB-624-connection-retry
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.5/bugfix/BB-624-connection-retry
```





*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-624-connection-retry` into
`development/7.10`
* [w/7.70/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/7.70...w/7.70/bugfix/BB-624-connection-retry) will be merged into `development/7.70` (pull request #2577)
* [w/8.5/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.5...w/8.5/bugfix/BB-624-connection-retry) will be merged into `development/8.5` (pull request #2583)
* [w/8.6/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.6...w/8.6/bugfix/BB-624-connection-retry) will be merged into `development/8.6` (pull request #2584)
* [w/8.7/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.7...w/8.7/bugfix/BB-624-connection-retry) will be merged into `development/8.7` (pull request #2585)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-624-connection-retry` into
`development/7.10`
* [w/7.70/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/7.70...w/7.70/bugfix/BB-624-connection-retry) will be merged into `development/7.70` (pull request #2586)
* [w/8.5/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.5...w/8.5/bugfix/BB-624-connection-retry) will be merged into `development/8.5` (pull request #2583)
* [w/8.6/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.6...w/8.6/bugfix/BB-624-connection-retry) will be merged into `development/8.6` (pull request #2584)
* [w/8.7/bugfix/BB-624-connection-retry](https://github.com/scality/backbeat/compare/development/8.7...w/8.7/bugfix/BB-624-connection-retry) will be merged into `development/8.7` (pull request #2585)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #211cf711ebc96e6e0b3653bb8216e3c22d6afaa8 on the integration branch
`w/8.6/bugfix/BB-624-connection-retry` is merging a branch which is neither the current
branch `bugfix/BB-624-connection-retry` nor the development branch
`development/8.6`.

It is likely due to a rebase of the branch `bugfix/BB-624-connection-retry` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #211cf711ebc96e6e0b3653bb8216e3c22d6afaa8 on the integration branch
`w/8.7/bugfix/BB-624-connection-retry` is merging a branch which is neither the current
branch `bugfix/BB-624-connection-retry` nor the development branch
`development/8.7`.

It is likely due to a rebase of the branch `bugfix/BB-624-connection-retry` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests, create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_pull_requests, create_integration_branches**

BourgoisMickael commented:
> Suggestions:
> 
> * add tests for "Prevent ProbeServer crash on startup by delaying it after the queue components are started".
> * add tests for Consumer service crashes in the event of a consumer error?

@nicolas2bert 
I added a simple test for the crash on probing, but for testing the whole task  file that stops on error it's not easily testable, I'll try to use `vm` module but it might take some time.

BourgoisMickael commented on extensions/notification/queueProcessor/QueueProcessor.js:
```diff
@@ -116,17 +116,20 @@ class QueueProcessor extends EventEmitter {
             next => this._setupNotificationConfigManager(next),
             next => this._setupDestination(this.destinationConfig.type, next),
-            next => this._destination.init(() => {
+            // if connection to destination fails, process will stop & restart
+            next => this._destination.init(next),
```
About this it looks like on zenko it could crash with a "callback already called" if destination Kafka is not up on startup

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_pull_requests**

BourgoisMickael commented:
Here is an Integration e2e run: https://github.com/scality/Integration/actions/runs/11699734386/job/32582300296

It should not trigger any exit implemented in this PR as all kafka dependencies are already UP in those tests

welansari commented on extensions/notification/queueProcessor/QueueProcessor.js:
```diff
@@ -142,22 +145,31 @@ class QueueProcessor extends EventEmitter {
+                    this.logger.error('error starting notification consumer',
+                    { method: 'QueueProcessor.start', error: err.message });
+                    // crash if got error at startup
+                    if (!this.isReady()) {
```
Should we use `this._consumer.isReady()` instead of adding a new method that does the same ?

BourgoisMickael replied:
I copied exactly the required changes coming from ZENKO branches so there would be no conflict with waterflow

BourgoisMickael replied:
It seems simpler like that

BourgoisMickael committed:
BB-624: Bump version 7.10.18

BourgoisMickael commented:
/create_integration_branches

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/7.70/bugfix/BB-624-connection-retry` with contents from `bugfix/BB-624-connection-retry`
and `development/7.70`.

Please resolve the conflict on **the integration branch** (`w/7.70/bugfix/BB-624-connection-retry`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/7.70/bugfix/BB-624-connection-retry
 $ git pull  # or "git reset --hard origin/w/7.70/bugfix/BB-624-connection-retry"
 $ git merge origin/development/7.70
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/bugfix/BB-624-connection-retry
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.70/bugfix/BB-624-connection-retry
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/8.5/bugfix/BB-624-connection-retry` with contents from `w/7.70/bugfix/BB-624-connection-retry`
and `development/8.5`.

Please resolve the conflict on **the integration branch** (`w/8.5/bugfix/BB-624-connection-retry`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/8.5/bugfix/BB-624-connection-retry
 $ git pull  # or "git reset --hard origin/w/8.5/bugfix/BB-624-connection-retry"
 $ git merge origin/development/8.5
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/7.70/bugfix/BB-624-connection-retry
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.5/bugfix/BB-624-connection-retry
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/8.6/bugfix/BB-624-connection-retry` with contents from `w/8.5/bugfix/BB-624-connection-retry`
and `development/8.6`.

Please resolve the conflict on **the integration branch** (`w/8.6/bugfix/BB-624-connection-retry`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/8.6/bugfix/BB-624-connection-retry
 $ git pull  # or "git reset --hard origin/w/8.6/bugfix/BB-624-connection-retry"
 $ git merge origin/development/8.6
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/8.5/bugfix/BB-624-connection-retry
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.6/bugfix/BB-624-connection-retry
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

BourgoisMickael commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/7.70`

* :heavy_check_mark: `development/8.5`

* :heavy_check_mark: `development/8.6`

* :heavy_check_mark: `development/8.7`



The following branches have **NOT** changed:

* `development/7.4`



Please check the status of the associated issue BB-624.

Goodbye bourgoismickael.




*The following options are set:* **approve, create_integration_branches**
